engine/src/main/java/org/apache/apex/engine/YarnAppLauncherImpl.java

```diff
@@ -142,8 +142,7 @@ public String getApplicationId()
     @Override
     public boolean isFinished()
     {
-      YarnClient yarnClient = YarnClient.createYarnClient();
-      try {
+      try (YarnClient yarnClient = StramClientUtils.createYarnClient(conf)) {
```

Comment:
Was wondering if there would be any issues holding the client for a long time such as stale connections etc, but following the code it seems to be ok to hold on to the client and reuse it.